pytensor/tensor/rewriting/elemwise.py

```diff
@@ -287,22 +286,17 @@ def create_inplace_node(self, node, inplace_pattern):
         op = node.op
         scalar_op = op.scalar_op
         inplace_pattern = {i: o for i, [o] in inplace_pattern.items()}
-        if hasattr(scalar_op, "make_new_inplace"):
-            new_scalar_op = scalar_op.make_new_inplace(
-                transfer_type(
-                    *[
-                        inplace_pattern.get(i, o.dtype)
-                        for i, o in enumerate(node.outputs)
-                    ]
+        try:
+            return type(op)(scalar_op, inplace_pattern).make_node(*node.inputs)
+        except TypeError:
+            # Elemwise raises TypeError if we try to inplace an output on an input of a different dtype
```

Comment:
No prior test caused this to happen, and I suspect it would be rare, but added an explicit handling and test.

In general it would be nice that `node.op.make_node(*node.inputs)` would always return the same type, but global config variables can change the behavior of `Scalar.make_node`. The `transfer_type` was working as a kind of `freeze_output_type` here I guess. We can revisit this approach later, but for now I'm inclined to go with simplicity.

Note that with the default `config.cast_policy="custom"` or `config.cast_policy="numpy"` floatX plays no role in `Scalar.make_node`, it only affects the initial creation of tensor/constants. Only `config.cast_policy="numpy+floatX"` plays a role here.